RapidReact/src/main/java/frc/robot/subsystem/ClimberSubsystem.java

```diff
@@ -1,112 +1,311 @@
+    if (autoClimb) return;
 
+    // TODO PERSISTNET: LIMIT SWITCHES https://docs.ctre-phoenix.com/en/stable/ch13_MC.html#limit-switches
+    // TODO PERSISTENT: you should have the joystick/ button move the motion magic setpoint, not the motor in PWM mode
 
-  public void fixedHookToggler() { //uses R1 button
-    if (config.enablePneumatics) {
-      fixedHookToggleState = !fixedHookToggleState;
-      if (!fixedHookToggleState) {
-        fixedHookSolenoid.set(Value.kForward);
-        hookToggleState.log(LogLevel.GENERAL, "fixedHookTime");
-      } else {
-        fixedHookSolenoid.set(Value.kReverse);
-        hookToggleState.log(LogLevel.GENERAL, "fixedHookTime");
-      }
-    }
-  }
-
-  public void elevatorExtend() { //uses up button
-    climber.set(ControlMode.PercentOutput, climbOutput.currentValue());
+    climber1.set(ControlMode.PercentOutput, climbOutput.currentValue());
+    climber2.set(ControlMode.PercentOutput, climbOutput.currentValue());
+    climberMotorPosition.log(climber1.getSelectedSensorPosition());
 
     climbState.log(LogLevel.GENERAL, "elevatorExtend");
   }
 
-  public void elevatorRetract() { //uses down button
-    climber.set(ControlMode.PercentOutput, -climbOutput.currentValue());
+  public void manualElevatorRetract() { //uses down button
+    if (autoClimb) return;
+
+    // TODO PERSISTNET: LIMIT SWITCHES https://docs.ctre-phoenix.com/en/stable/ch13_MC.html#limit-switches
+    // TODO PERSISTENT: you should have the joystick/ button move the motion magic setpoint, not the motor in PWM mode
+
+    climber1.set(ControlMode.PercentOutput, -climbOutput.currentValue());
+    climber2.set(ControlMode.PercentOutput, -climbOutput.currentValue());
+    climberMotorPosition.log(climber1.getSelectedSensorPosition());
 
     climbState.log(LogLevel.GENERAL, "elevatorRetract");
   }
 
   public void elevatorStop() {
-    climber.set(0);
+    climber1.set(0);
+    climber2.set(0);
     climbState.log(LogLevel.GENERAL, "climbStopped");
   }
 
-  public void climbAuto() { //uses TPAD button
+  // autoClimb() both enables autoclimb and sets the flag to true
+  // autoClimbReleased() just sets the flag to false
+  public void autoClimb() { //uses TPAD button
     autoClimb = true;
-    climbState.log(LogLevel.GENERAL, "climb2auto");
+    autoClimbPressed = true;
+
+    climbState.log(LogLevel.GENERAL, "automatically climbing");
+  }
+  public void autoClimbReleased()
+  {
+    autoClimbPressed = false;
   }
 
-  public void elevatorToggle() {
-    if (config.enablePneumatics) {
-      elevatorToggle = !elevatorToggle;
-      if (!elevatorToggle) {
-        elevatorSolenoid.set(Value.kForward);
-        elevatorToggleState.log(LogLevel.GENERAL, "ClimbTime");
-      } else {
-        elevatorSolenoid.set(Value.kReverse);
-        elevatorToggleState.log(LogLevel.GENERAL, "NotClimbTime");
-      }
+  private void autoExtendPartial()
+  {
+    climber1.set(TalonSRXControlMode.MotionMagic, partialExtendPosition);
```

Comment:
Don't forget to add a secondary control loop synchronizing the positions: https://docs.ctre-phoenix.com/en/stable/ch16_ClosedLoop.html#example-2-lift-mechanism